sdk/python/kfp/compiler/_container_builder.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import tarfile
+import tempfile
+import os
+import uuid
+
+class ContainerBuilder(object):
+  """
+  ContainerBuilder helps build a container image
+  """
+  def __init__(self, gcs_staging):
```

Comment:
I think the builder constructor should also accept namespace and timeout.

Reply:
Are you suggesting the constructor accepting the namespace and timeout as the default value if no such params are provided when the build function is called?

Reply:
> Are you suggesting the constructor accepting the namespace and timeout as the default value if no such params are provided when the build function is called?
Maybe the `build` function does not need those parameters).

This way you can create a different builder (e.g. `LocalDockerBuilder` or `CloudBuildBuilder`) with the same `build(context_dir, target_image, docker_filename="./Dockerfile")` interface.

Reply:
I like the idea. Will convert it that way.